code/script.js

```diff
@@ -229,97 +237,171 @@ const setSecret = () => {
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label // This variable stores what option group (category) the question belongs to.
+  const value = questions.value // We also need a variable that stores the actual value of the question we've selected.
 
-  currentQuestion = {
+  currentQuestion = { // currentQuestion object created, before it was just a variable that we declared
     category: category,
-    // value: value
+    value: value,
   }
+  console.log(currentQuestion)
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+    if (category === 'hair' || category === 'eyes') {
+      if (secret[category] === value) {
+        filterCharacters(true)
+      } 
+      else {
+        filterCharacters(false)
+      }
+       
+    } else if (category === 'accessories' || category === 'other') {
+        if (secret[category].includes(value)) {
+          filterCharacters(true)
+        }
+        else {
+          filterCharacters(false)
+        }
+      } 
+}        
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+  //Alternative way for checkQuestion
+  //  let keep = false
+    
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  //   if (category === 'hair' || category === 'eyes') {
+  //     keep = (value === secret[category])
+  //     console.log(keep)
 
-  } else if (category === 'accessories' || category === 'other') {
+  //   } else if (category === 'accessories' || category === 'other') {
+  //     keep = secret[category].includes(value)
+  //   }
 
-  }
-}
+  //   filterCharacters(keep)
+  // }
+  
+
+// Filter the characters array and redraw the game board.
 
-// It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
+
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+  
+  if (category === 'hair') {
+    if(keep) {
+      modal.style.display = 'block' //modal window shown
+      modalText.innerHTML = `
+      Yes, the person has ${value} ${category}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+
```

Comment:
Category hair and eyes have same text that you're showing to the user, I wonder if it could be better to combine those two in one if statement so your code can be a lit shorter and not reaping itself :)